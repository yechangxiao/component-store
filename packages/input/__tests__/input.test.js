import Input from '../src/input.vue'
import { mount } from '@vue/test-utils'

describe('rb-input', () => {
  test('input-text', () => {
    const wrapper = mount(Input)
    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-password', () => {
    const wrapper = mount(Input, {
      propsData: {
        type: 'password'
      }
    })
    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-value', () => {
    const wrapper = mount(Input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.props('value')).toBe('admin')
  })

  test('input-snapshot', () => {
    const wrapper = mount(Input, {
      propsData: {
        type: 'text',
        value: 'admin'
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})