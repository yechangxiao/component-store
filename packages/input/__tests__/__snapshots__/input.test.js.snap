// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rb-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
